documentation/library_guide/cmake_support.rst

```diff
@@ -0,0 +1,109 @@
+
+For more details on |onedpl_short| backends, see :doc:`Execution Policies <parallel_api/execution_policies>`.
+
+Example CMake File
+==================
+To use |onedpl_short| with CMake, a user must create a ``CMakeLists.txt`` file for their project and add |onedpl_short|.  This file should be placed in the project's base directory.  Below is an example ``CMakeLists.txt`` file:
+
+.. code:: cpp
+
+  if (CMAKE_HOST_WIN32)
+    find_package(oneDPLWindowsIntelLLVM)
+  endif()
+
+  project(Foo)
+  add_executable(foo foo.cpp)
+  
+  # Search to find oneDPL
+  find_package(oneDPL REQUIRED)
+  
+  # Connect oneDPL to foo
+  target_link_libraries(foo oneDPL)
+
+**Note:** On Windows, some workarounds may be required to use icx[-cl] successfully with |onedpl_short|.  We recommend updating to the most recent version of CMake to minimize the workarounds required for successful use.  A CMake package has been provided, ``oneDPLWindowsIntelLLVM``, to provide the necessary workarounds to enable support for icx[-cl] on Windows with CMake versions 3.20 and greater.  Some workarounds are provided for icpx, but it is not fully supported on Windows at this time.  To use this package, please add ``find_package(oneDPLWindowsIntelLLVM)`` to your CMake file *before* you call ``project()``.
+
+
+Example CMake Invocation
+========================
+After creating a ``CMakeLists.txt`` file for their project, a user may use a command line CMake invocation to generate build scripts for their project.
+
+Below is an example ``Linux`` CMake invocation which generates Unix makefiles for the project with the ``icpx`` compiler, ``tbb`` backend and ``release`` build type:
+
+.. code:: cpp
+
+  mkdir build && cd build
+  cmake -DCMAKE_CXX_COMPILER=icpx -DCMAKE_BUILD_TYPE=release -DONEDPL_PAR_BACKEND=tbb ..
+
+Below is an example ``Windows`` CMake invocation which generates ``Ninja`` build scripts (see :ref:`Requirements Section<Requirements>`) for the project in the parent directory with the ``icx`` compiler, ``OpenMP`` backend and ``debug`` build type:
+
+.. code:: cpp
+
+  mkdir build && cd build
+  cmake -GNinja -DCMAKE_CXX_COMPILER=icx -DCMAKE_BUILD_TYPE=debug -DONEDPL_PAR_BACKEND=openmp ..
+
+Both of these examples assume the starting working directory is the project's base directory which contains ``CMakeLists.txt``.  The build scripts are generated in a newly created ``build`` directory.
+
+
+Example Build command
+=====================
+Once build scripts have been generated for your desired configuration following the instruction above, a `build command <https://cmake.org/cmake/help/latest/manual/cmake.1.html#build-a-project>`_ can be issued to build your project:
+
+.. code:: cpp
+
+  cmake --build .
+
+This example assumes the starting working directory is in the directory which contains the CMake generated build scripts, ``build``, if following the instructions above.
+
+
+Requirements
```

Comment:
Could you place this section between "General Usage" and "Backend Options"? I think it should precede the detailed description of the APIs.

Reply:
Done.